Lib/test/test_typing.py

```diff
@@ -7442,9 +7433,9 @@ def asend(self, value):
             def athrow(self, typ, val=None, tb=None):
                 pass
 
-        ns = {}
-        exec('async def g(): yield 0', globals(), ns)
-        g = ns['g']
+        async def g(): yield 0
+        g = g()
```

Comment:
In the old code here g was the function, now it's the coroutine returned from the function.

Reply:
good catch. Sorry, that was sloppy.

Reply:
Although on second thought... doesn't it make more sense to test the coroutine returned from the function here? I think the test might be buggy as-written.

It passes regardless, and this PR wasn't meant to make any semantic changes to the tests, so I'm happy to leave it as-is for now, but it looks odd to me

Reply:
It makes sense that it passes both with and without the `g()`, because `g` is only used to check that the [coroutine-function or coroutine] is _not_ an instance of the custom `AsyncGenerator` subclass here: https://github.com/python/cpython/blob/7d7eec595a47a5cd67ab420164f0059eb8b9aa28/Lib/test/test_typing.py#L7459-L7460

And of course, it is indeed true that neither a coroutine nor a coroutine function will be an instance of the custom `AsyncGenerator` subclass